### ISSUE TITLE
not installable on ghc 9.0.1 due to reliance on base <4.15

### ISSUE THREAD

**Issue by stuart-little** [mrkkrp/req#118]
I am running:

```
$ ghc --version             
The Glorious Glasgow Haskell Compilation System, version 9.0.1
```
and
```
$ cabal --version           
cabal-install version 3.4.0.0
```
On that setup, trying to install `req` with
```
cabal install --lib req
```
returns this:
```
Resolving dependencies...
cabal: Could not resolve dependencies:
[__0] trying: base-4.15.0.0/installed-4.15.0.0 (user goal)
[__1] trying: req-3.9.0 (user goal)
[__2] trying: http-client-tls-0.3.5.3 (dependency of req)
[__3] trying: memory-0.15.0 (dependency of http-client-tls)
[__4] trying: memory:+support_foundation
[__5] next goal: basement (dependency of memory +support_foundation)
[__5] rejecting: basement-0.0.11 (conflict: base==4.15.0.0/installed-4.15.0.0,
basement => base>=4.9.0.0 && <4.15)
[__5] skipping: basement-0.0.10, basement-0.0.9, basement-0.0.8,
basement-0.0.7, basement-0.0.6, basement-0.0.5, basement-0.0.4,
basement-0.0.3, basement-0.0.2, basement-0.0.1, basement-0.0.0 (has the same
characteristics that caused the previous version to fail: excludes 'base'
version 4.15.0.0)
[__5] fail (backjumping, conflict set: base, basement, memory,
memory:support_foundation)
After searching the rest of the dependency tree exhaustively, these were the
goals I've had most trouble fulfilling: memory, base, foundation,
memory:support_foundation, basement, http-client-tls, req
Try running with --minimize-conflict-set to improve the error message.
```
So `req` ultimately depends on [basement](https://hackage.haskell.org/package/basement), which currently requires `base <4.15`. Since `base` is exactly at `4.15` in `ghc 9.0.1`, the conflict ensues.

Is this just a matter of waiting for `basement` to catch up, or is there some other solution?

**Comment by mrkkrp** [comment]
Basement holds us back here.

**Comment by stuart-little** [comment]
Yeah.. After more digging I noticed `ghc 9.0.1` support is very much under active discussion [there](https://github.com/haskell-foundation/foundation/issues/551) (see also [this](https://github.com/haskell-foundation/foundation/pull/549) recently-closed pull request).

So I suppose there's not much to do but wait: no point reporting to the `basement` folks either, since they seem to know about the problems and are working on them.

**Comment by mrkkrp** [comment]
Fixed by #117. I'm going to release a new version on Hackage.
